frontend/src/metabase/dashboard/components/EmbedMenu/AdminEmbedMenu.tsx

```diff
@@ -53,8 +56,22 @@ export const AdminEmbedMenu = ({
     );
   }
 
+  const onMenuToggle = (t: boolean) => {
+    if (t && resourceType === "question" && !resource.isSaved()) {
+      onModalOpen(MODAL_TYPES.SAVE_QUESTION_BEFORE_EMBED);
```

Comment:
I don't like this solution. It makes this component tightly coupled with how it should be used, ultimately, in `ViewFooter` or `DashboardEmbedAction` and this makes the `onModalOpen` API inconsistent which creates confusion.

If you look in `ViewFooter` (query builder)
https://github.com/metabase/metabase/blob/ace3337474ef1860e78d25b95bebd7622b3a22fd/frontend/src/metabase/query_builder/components/view/ViewFooter.jsx#L151

`onModalOpen` = (modalMode) => void

but in `DashboardEmbedAction`
https://github.com/metabase/metabase/blob/ace3337474ef1860e78d25b95bebd7622b3a22fd/frontend/src/metabase/dashboard/components/DashboardEmbedAction/DashboardEmbedAction.tsx#L22
`onModalOpen` = (void) => void

Also, `onMenuToggle` here _conditonally_ calls `onModalOpen` which makes it hard to understand when my `onModalOpen` would be called or not.

Could you simplify this part more? I'd expect the `onModalOpen` API to be more consistent, and don't have this kind of condition inside the button. The components that use `EmbedMenu` should be the one that tells which modal to open instead of doing that in this component

Reply:
Yep, I felt uncomfortable writing this solution as well. You make a good point about the consistency of the `onModalOpen` call, I'll see if I can make it consistent across the different components